pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -649,7 +649,8 @@ func (p *criStatsProvider) getAndUpdateContainerUsageNanoCores(stats *runtimeapi
 		defer p.mutex.Unlock()
 
 		cached, ok := p.cpuUsageCache[id]
-		if !ok || cached.stats.UsageCoreNanoSeconds == nil {
+		cpuAcctReset := ok && stats.Cpu.UsageCoreNanoSeconds.Value < cached.stats.UsageCoreNanoSeconds.Value
+		if !ok || cached.stats.UsageCoreNanoSeconds == nil || cpuAcctReset {
```

Comment:
You need to check if `UsageCoreNanoSeconds == nil` before comparing the values.

Reply:
Since you are already checking `cached.stats.UsageCoreNanoSeconds == nil` and `ok` in the if statement, it is probably just easier to do the comparison at the end of the `if` statement, rather than before...

Reply:
haha lets revert this back to the way it was before. `cpuAcctReset` was requested to be added for clarity, but I think the logic is cleaner with this just added to the if statement because of the resulting duplication of checks.